src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -83,7 +83,7 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
             if (LambdaUtilities.IsLambdaBody(syntax))
             {
                 Debug.Assert(allowLambda);
-                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax);
+                Debug.Assert(syntax is ExpressionSyntax || syntax is BlockSyntax || syntax is ArrowExpressionClauseSyntax);
```

Comment:
Not for EnC analysis. EnC has very specific concept of "body".  

---
In reply to: [132526663](https://github.com/dotnet/roslyn/pull/21396#discussion_r132526663) [](ancestors = 132526663)